### PR TITLE
151 vnv nfrs 101 134

### PR DESCRIPTION
**Pull request by OlegGlo**

### CODE REVIEW

File: docs/VnVPlan/VnVPlan.tex [hunk 1]
```diff
@@ -316,6 +316,1047 @@ \subsection{Tests for Nonfunctional Requirements}
...
+Input/Condition: Users interact with the platform exclusively using keyboard navigation without the use of a mouse.
+
+How test will be performed:
+\begin{enumerate}
+    \item \textbf{Identify Core Actions:}
+    \begin{enumerate}
+        \item Compile a list of all core actions within the platform (e.g., navigating between tasks, submitting labels, accessing settings).
+        \item Document the corresponding keyboard shortcuts for each core action.
+    \end{enumerate}
+    \item \textbf{User Recruitment:}
```
Comment (matpetro) [inline docs/VnVPlan/VnVPlan.tex:620]: Do we really need User tesing and recruitment for this one? A tester on our team can just test the shortcuts

File: docs/VnVPlan/VnVPlan.tex [hunk 2]
```diff
@@ -316,6 +316,1047 @@ \subsection{Tests for Nonfunctional Requirements}
...
+
+Type: Automated, Performance
+
+Initial State: The system is operational and capable of accepting new user account creation requests.
+
+Input/Condition: A large sample of new user account creation requests are submitted to the system.
+
+How test will be performed:
+\begin{enumerate}
+    \item \textbf{Request Generation:}
```
Comment (matpetro) [inline docs/VnVPlan/VnVPlan.tex:666]: I don't think all these Sub titles are necessary for the test, just list them out in one itemized list

File: docs/VnVPlan/VnVPlan.tex [hunk 3]
```diff
@@ -316,6 +316,1047 @@ \subsection{Tests for Nonfunctional Requirements}
...
+
+Type: Automated, Performance
+
+Initial State: The system is operational and capable of handling service requests and generating reports for Customers.
+
+Input/Condition: A large sample of service requests with varying negotiated time limits are submitted to the system.
+
+How test will be performed:
+\begin{enumerate}
+    \item \textbf{Request Generation:}
```
Comment (matpetro) [inline docs/VnVPlan/VnVPlan.tex:705]: Once again, all these sub titles not really needed, just make into one list. This is applicable to a few of your tests so please read them over.

**Review by matpetro — CHANGES_REQUESTED** [review]
Looks okay but here are some main points of improvement that should be implemented:
- For tests that have sub titles/ headings for "how test will be performed", a lot of them aren't necessary, just make it into one list
- Whenever you have a measurement of some kind (percent, timing etc.) that is explicitly a number, that number should be put in section 6 (Symbolic Parameters) and referenced accordingly
- I noticed no tests have an output section, I would add that in where its appropriate
- Consistency in Initial State and input sections: some are described in sentences, others are in bullet point
- Add the necessary Usability Survey Questions that relate to your tests

File: docs/VnVPlan/VnVPlan.tex [hunk 4]
```diff
@@ -316,6 +316,1047 @@ \subsection{Tests for Nonfunctional Requirements}
...
+\end{enumerate}
+
+% NFRs 12---------------------------------------------------------
+\subsubsection{Performance Requirements}
+% 12.1
+\paragraph{Speed and Latency Tests}
+\begin{enumerate}
+\item{New User Account Processing Time Validation: T-PR0\\}
+
+Related Reqs: NFR-PR0
```
Comment (matpetro) [inline docs/VnVPlan/VnVPlan.tex:656]: I think this relates to both PR0 and PR1 correct?

File: docs/VnVPlan/VnVPlan.tex [hunk 5]
```diff
@@ -316,6 +316,1047 @@ \subsection{Tests for Nonfunctional Requirements}
...
+    \end{enumerate}
+    \item \textbf{Fit Criterion Verification:}
+    \begin{enumerate}
+        \item Compute the percentage of requests processed within 0.25 hours (15 minutes).
+        \item Verify that at least 90\% of the requests meet this criterion.
+        \item Additionally, ensure that 100\% of the requests are processed within 48 hours.
+    \end{enumerate}
+\end{enumerate}
+
+\item{Service Request Report Timing Validation: T-PR2\\}
```
Comment (matpetro) [inline docs/VnVPlan/VnVPlan.tex:693]: You skipped from T-PR0 to T-PR2

File: docs/VnVPlan/VnVPlan.tex [hunk 6]
```diff
@@ -316,6 +316,1047 @@ \subsection{Tests for Nonfunctional Requirements}
...
+    \item \textbf{Model Deployment:}
+    \begin{enumerate}
+        \item Deploy the trained models to the production environment using each framework's deployment tools.
+        \item Ensure that the deployment process completes successfully and that the models are accessible for inference.
+    \end{enumerate}
+    
+\end{enumerate}
+
+\end{enumerate}
+
```
Comment (matpetro) [inline docs/VnVPlan/VnVPlan.tex:1314]: I think we will need a test to cover NFR-OE5

File: docs/VnVPlan/VnVPlan.tex (hunk 1 above)
Comment (OlegGlo) [reply docs/VnVPlan/VnVPlan.tex:620]: agreed

File: docs/VnVPlan/VnVPlan.tex (hunk 4 above)
Comment (OlegGlo) [reply docs/VnVPlan/VnVPlan.tex:656]: Yup

File: docs/VnVPlan/VnVPlan.tex (hunk 5 above)
Comment (OlegGlo) [reply docs/VnVPlan/VnVPlan.tex:693]: PR1 is the same as PR0 functionally

**Review by matpetro — APPROVED** [review]
Looks good

**Pull request merged**